### PR TITLE
Notification

### PR DESCRIPTION
**Pull request by BykhovDenis**

### CODE REVIEW

File: packages/core/src/predicate.ts [hunk 1]
```diff
@@ -43,6 +43,28 @@ const predicates: Record<string, PredicateFactory> = {
...
       return result
     }
+  },
+
+  $gt: (query: number, propertyKey: string): Predicate => {
+    return (docs: Doc[]): Doc[] => {
+      const result: Doc[] = []
+      for (const doc of docs) {
+        const value = (doc as any)[propertyKey]
+        if (value - query > 0) result.push(doc)
```
Comment (haiodo) [inline packages/core/src/predicate.ts:53]: Should it be just for number formats?

File: packages/query/src/index.ts [hunk 2]
```diff
@@ -134,6 +135,10 @@ export class LiveQuery extends TxProcessor implements Storage, Queriable {
       const updatedDoc = q.result.find((p) => p._id === tx.objectId)
       if (updatedDoc !== undefined) {
         await this.doUpdateDoc(updatedDoc, tx)
+        if (!this.match(q, updatedDoc)) {
+          q.result.splice(0, q.result.indexOf(updatedDoc))
+          q.total--
```
Comment (haiodo) [inline packages/query/src/index.ts:140]: good catch

File: plugins/chunter-impl/src/components/ChannelView.svelte [hunk 3]
```diff
@@ -37,8 +39,10 @@
   let query: QueryUpdater<Message> | undefined
 
   $: if (currentSpace !== undefined) {
-    query = client.query(query, chunter.class.Message, { space: currentSpace }, (result) => {
+    query = client.query(query, chunter.class.Message, { space: currentSpace }, async (result) => {
       messages = result
+      const notificationP = await getPlugin(notificationPlugin.id)
```
Comment (haiodo) [inline plugins/chunter-impl/src/components/ChannelView.svelte:44]: Please move to constants

File: plugins/notification-impl/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+  const accountId = client.accountId()
+
+  const markAsRead = async <T extends Doc>(objectClass: Ref<Class<T>>, space: Ref<Space>): Promise<void> => {
+    const query = {
+      objectClass: objectClass,
+      space: space,
+      client: accountId
+    }
+    // eslint-disable-next-line no-void
+    void client.findAll(notification.class.LastViewed, query).then((subscribes) => {
```
Comment (haiodo) [inline plugins/notification-impl/src/index.ts:36]: why not use await for client.findAll?

File: packages/core/src/predicate.ts (hunk 1 above)
Comment (BykhovDenis) [reply packages/core/src/predicate.ts:53]: Now, yes, I can hardly imagine a situation when you need `gt` not by the number ...

File: plugins/notification-impl/src/index.ts (hunk 4 above)
Comment (BykhovDenis) [reply plugins/notification-impl/src/index.ts:36]: Faster

File: plugins/chunter-impl/src/components/ChannelView.svelte (hunk 3 above)
Comment (BykhovDenis) [reply plugins/chunter-impl/src/components/ChannelView.svelte:44]: Not understood. What exactly should I move?

**Pull request closed without merging**